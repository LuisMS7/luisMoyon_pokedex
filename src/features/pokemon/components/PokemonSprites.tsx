import React, { useCallback, useMemo } from "react";
import { Image, ScrollView, StyleSheet, View } from "react-native";
import { perfectSize } from "@utils/perfect-size";
import SvgUri from "react-native-svg-uri";
import { Sprites } from "@pokemon/types/pokemon-api-response.type";
// eslint-disable-next-line max-len
import extractStringsFromObject from "@pokemon/utils/extract-string-from-object";

type PokemonSpriteProps = {
	imgUrl: string;
};

const PokemonSprite = ({ imgUrl }: PokemonSpriteProps) => {
	const getPokemonImg = useCallback(() => {
		if (
			imgUrl.endsWith("jpg") ||
			imgUrl.endsWith("png") ||
			imgUrl.endsWith("gif")
		) {
			return (
				<Image
					source={{ uri: imgUrl }}
					style={{
						height: perfectSize(60),
						width: perfectSize(60),
					}}
					resizeMode={"cover"}
				/>
			);
		} else if (imgUrl.endsWith("svg")) {
			return (
				<SvgUri
					source={{ uri: imgUrl }}
					height={perfectSize(60)}
					width={perfectSize(60)}
				/>
			);
		}
		return null;
	}, [imgUrl]);
	return <View style={{ ...styles.stripeContainer }}>{getPokemonImg()}</View>;
};

type PokemonSpritesProps = {
	sprites: Sprites;
};
const PokemonSprites = ({ sprites }: PokemonSpritesProps) => {
	const spritesImages = useMemo(() => {
		return extractStringsFromObject(sprites);
	}, [sprites]);

	return (
		<ScrollView contentContainerStyle={styles.container} horizontal>
			{spritesImages.map((sprite, id) => (
				<PokemonSprite imgUrl={sprite} key={`sprite-${id}`} />
			))}
		</ScrollView>
	);
};

const styles = StyleSheet.create({
	container: {
		flexDirection: "row",
		gap: perfectSize(10),
		alignSelf: "center",
	},
	stripeContainer: {
		borderRadius: 1000,
		height: perfectSize(80),
		width: perfectSize(80),
		alignItems: "center",
		justifyContent: "center",
	},
});

export default PokemonSprites;
