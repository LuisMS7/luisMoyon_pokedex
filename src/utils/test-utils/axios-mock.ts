import client from "@lib/axios-instance";
import MockAdapter from "axios-mock-adapter";

export const mockClient = new MockAdapter(client);

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const mockGetOnSuccess = (endpoint: string, mockResponse: any) => {
	mockClient.onGet(endpoint).reply(200, mockResponse);
};

export const mockGetOnFailure = (endpoint: string, statusCode: number) => {
	mockClient.onGet(endpoint).reply(statusCode);
};
