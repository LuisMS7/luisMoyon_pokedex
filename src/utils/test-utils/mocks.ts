import { Pokemon } from "@pokemon/types/pokemon.type";

export const mockPokemon: Pokemon = {
	id: 1,
	name: "mockPokemonName",
	// eslint-disable-next-line max-len
	img: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png",
	types: [{ type: { name: "normal", url: "" }, slot: 1 }],
	weight: 10,
	sprites: {
		back_default:
			// eslint-disable-next-line max-len
			"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png",
		back_female: "",
		back_shiny: "",
		back_shiny_female: "",
		front_default: "",
		front_female: "",
		front_shiny: "",
		front_shiny_female: "",
	},
	moves: [{ move: { name: "normal", url: "" }, version_group_details: [] }],
};
