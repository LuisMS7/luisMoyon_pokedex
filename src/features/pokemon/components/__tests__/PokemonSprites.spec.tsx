import React from "react";
import { render, screen } from "@testing-library/react-native";
import { mockPokemon } from "@utils/test-utils/mocks";
import PokemonSprites from "@pokemon/components/PokemonSprites";

describe("PokemonSprites", () => {
	it("should render correctly", () => {
		render(<PokemonSprites sprites={mockPokemon.sprites!} />);

		expect(screen.toJSON()).toMatchSnapshot();
	});
});
