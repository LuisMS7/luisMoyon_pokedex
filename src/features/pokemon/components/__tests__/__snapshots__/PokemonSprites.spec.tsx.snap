// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonSprites should render correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "alignSelf": "center",
      "flexDirection": "row",
      "gap": 14.909958023215566,
    }
  }
  horizontal={true}
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "borderRadius": 1000,
          "height": 119.27966418572453,
          "justifyContent": "center",
          "width": 119.27966418572453,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          {
            "uri": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png",
          }
        }
        style={
          {
            "height": 89.4597481392934,
            "width": 89.4597481392934,
          }
        }
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "borderRadius": 1000,
          "height": 119.27966418572453,
          "justifyContent": "center",
          "width": 119.27966418572453,
        }
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "borderRadius": 1000,
          "height": 119.27966418572453,
          "justifyContent": "center",
          "width": 119.27966418572453,
        }
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "borderRadius": 1000,
          "height": 119.27966418572453,
          "justifyContent": "center",
          "width": 119.27966418572453,
        }
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "borderRadius": 1000,
          "height": 119.27966418572453,
          "justifyContent": "center",
          "width": 119.27966418572453,
        }
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "borderRadius": 1000,
          "height": 119.27966418572453,
          "justifyContent": "center",
          "width": 119.27966418572453,
        }
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "borderRadius": 1000,
          "height": 119.27966418572453,
          "justifyContent": "center",
          "width": 119.27966418572453,
        }
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "borderRadius": 1000,
          "height": 119.27966418572453,
          "justifyContent": "center",
          "width": 119.27966418572453,
        }
      }
    />
  </View>
</RCTScrollView>
`;
